db/version_set.cc

```diff
@@ -1875,13 +1875,25 @@ void Version::Get(const ReadOptions& read_options, const LookupKey& k,
   bool is_blob_index = false;
   bool* const is_blob_to_use = is_blob ? is_blob : &is_blob_index;
 
+  auto get_blob_callback = [this, &is_blob_index, &read_options, &user_key,
+                            &value](const Slice& blob_index,
+                                    Slice& blob_value) -> Status {
+    Status s;
+    if (is_blob_index) {
+      constexpr uint64_t* bytes_read = nullptr;
+      s = GetBlob(read_options, user_key, blob_index, value, bytes_read);
+      blob_value = static_cast<Slice>(*value);
+    }
+    return s;
+  };
+
```

Comment:
Love the callback idea but instead of using `std::function` and a lambda, can we use a concrete class? (Rationale: `std::function` does type erasure, which has similar runtime overhead as virtuality.)